File: test/js/case19_try_raise.js
// このファイルは自動生成です（Python -> JavaScript native mode）。

const __pytra_root = process.cwd();
const py_runtime = require(__pytra_root + '/src/js_module/py_runtime.js');
const py_math = require(__pytra_root + '/src/js_module/math.js');
const py_time = require(__pytra_root + '/src/js_module/time.js');
const { pyPrint, pyLen, pyBool, pyRange, pyFloorDiv, pyMod, pyIn, pySlice, pyOrd, pyChr, pyBytearray, pyBytes, pyIsDigit, pyIsAlpha } = py_runtime;
const { perfCounter } = py_time;

function maybe_fail_19(flag) {
    try {
        if (pyBool(flag)) {
            throw Exception('fail-19');
        }
        return 10;
    }
    catch (ex) {
        return 20;
    }
    finally {
    }
}
pyPrint(maybe_fail_19(true));

File: test/fixtures/py/control/case19_try_raise.py
# このファイルは `test/fixtures/py/case19_try_raise.py` のテスト/実装コードです。
# 役割が分かりやすいように、読み手向けの説明コメントを付与しています。
# 変更時は、既存仕様との整合性とテスト結果を必ず確認してください。


from py_module.py_runtime import py_assert_stdout
def maybe_fail_19(flag: bool) -> int:
    try:
        if flag:
            raise Exception("fail-19")
        return 10
    except Exception as ex:
        return 20
    finally:
        pass


def _case_main() -> None:
    print(maybe_fail_19(True))

if __name__ == "__main__":
    print(py_assert_stdout(['20'], _case_main))
